### ISSUE TITLE
Update FIPS compliance mode to allow TLS 1.3

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#52926]
The "new" boringcrypto module has been validated: https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4735

* [x] Envoy updates the library https://github.com/envoyproxy/envoy/pull/35534.
* [ ] Envoy sets `set_compliance_policy` (or we find its not needed)
* [x] Go https://github.com/golang/go/issues/62372. Update merged in https://go-review.googlesource.com/c/go/+/603375 (NOT in go1.23, so presumably in go1.24
  * [x] Go sets ` set_compliance_policy` or equivilent https://go-review.googlesource.com/c/go/+/603376
* [ ] Rust boring wrapper (which we can optionally build for Ztunnel): https://github.com/cloudflare/boring/pull/214

**Comment by keithmattix** [comment]
My vote would be new compliance policy

**Comment by howardjohn** [comment]
cc @keithmattix @kyessenov @pmerrison

**Comment by kyessenov** [comment]
New policy sounds fine to me. set_compliance_policy is a must, otherwise, the restrictions are not effective.

**Comment by pmerrison** [comment]
I agree that we should have a new one simply because a user currently sets COMPLIANCE_POLICY to FIPS-140-2, and the new module is 140-3 validated. Reusing the old one will probably cause user confusion.

**Comment by dhawton** [comment]
+1 on a new policy.

**Comment by danielewood** [comment]
Of note, the new [golang 1.24 release](https://tip.golang.org/doc/security/fips140) makes FIPS mode a first class citizen in golang.

**Comment by gil-tohar-cyera** [comment]
Is there any expectation for when this will be available? Without this, Istio Ambient Mode cannot be used with FIPS, since ztunnel requires TLS 1.3.

cc: @keithmattix @kyessenov @pmerrison @howardjohn

**Comment by howardjohn** [comment]
This is blocked by https://github.com/envoyproxy/envoy/issues/31746 which has some more info.

Normally I wouldn't mention vendor features on github, but since this is an issue about non-OSS features anyways figured I would mention - @gil-tohar-cyera we (Solo.io) have FIPS ambient builds if you are interested.

**Comment by gil-tohar-cyera** [comment]
Thanks for the reference, we will look into it.

**Comment by howardjohn** [comment]
https://github.com/envoyproxy/envoy/issues/39056 gives the envoy side support

**Comment by gil-tohar-cyera** [comment]
Hi @howardjohn do you know if this is still in progress now that the envoy side isn't blocking?

**Comment by tvalasek** [comment]
Hi. Is there update on this?
